### PR TITLE
Limit AI Retries

### PR DESCRIPTION
**Pull request by officiallygod**
## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [x] Linting and tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by officiallygod** [comment]
Onboarding Data: https://github.com/newfold-labs/wp-module-onboarding-data/pull/83

File: src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js
Comment (ajayadav09) [inline src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js:1]: Disable retry on Mobile:

Screenshot 2024-05-16 at 11 07 47 PM

File: src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js [hunk 1]
```diff
@@ -55,6 +56,10 @@ const SiteGenStepErrorState = () => {
 		setIsHeaderNavigationEnabled( true );
 		setDrawerActiveView( false );
 		setSidebarActiveView( false );
+		setDisableRetry(
+			currentData.sitegen?.siteGenErrorMeta?.retryCount >=
+				currentData.sitegen?.siteGenErrorMeta?.maxRetryCount
+		);
```
Comment (ajayadav09) [inline src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js:62]: The retryCount when sitemeta apis fail exceeds more than 3 so the button is already disabled in the error screen. 

Screenshot 2024-05-16 at 11 04 05 PM

**Review by ajayadav09 — COMMENTED** [review]
the retryCount exceeds the maxRetryCount all the 9 apis fail in the first attempt.

File: src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js
Comment (officiallygod) [reply src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js:1]: My Bad Sure, Thanks.

File: src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js (hunk 1 above)
Comment (officiallygod) [reply src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js:62]: Yeah, I was thinking as every call costs us, would it be right to go forward considering all of them are failing.

Comment (officiallygod) [reply src/OnboardingSPA/components/ErrorState/Step/SiteGen/index.js:62]: Fixed this please check, Thanks.

**Comment by officiallygod** [comment]
> the retryCount exceeds the maxRetryCount all the 9 sitemeta apis fail in the first attempt.

Made sure all of that counts as one, Thanks.

**Review by ajayadav09 — APPROVED** [review]
Looks good for QA

**Comment by ajayadav09** [comment]
Verified changes for https://jira.newfold.com/browse/PRESS0-1370

**Pull request merged**